frontend/src/Editor/QueryManager/QueryEditors/TooljetDatabase/util.js

```diff
@@ -20,15 +20,15 @@ import { get } from 'lodash';
  *   },
  * };
  *
- * const result = hasEqualWithNull(queryOptions); // true
+ * const result = hasEmptyStringOrNullValue(queryOptions); // true
  */
-export const hasEqualWithNull = (queryOptions, operation) => {
+export const hasEmptyStringOrNullValue = (queryOptions, operation) => {
   const filters = get(queryOptions, `${operation}.where_filters`);
   if (filters) {
     const filterKeys = Object.keys(filters);
     for (let i = 0; i < filterKeys.length; i++) {
       const filter = filters[filterKeys[i]];
-      if (filter.operator !== 'is' && filter.value === null) {
+      if (filter.operator !== 'is' && (filter.value === '' || filter.value === null)) {
```

Comment:
Empty string can be input to the column. The fix doesn't allow to match a empty string in the column. For `is operator` the input can be empty either null or not null. @akshaysasidrn Could you correct me if the explanation is wrong.

Reply:
@ganesh8056 yeah you are right here. We need to validate for empty filters before executing the query.